Problematic outcome from `AddDate(0,0,0)` and daylight saving time

As noted in https://github.com/golang/go/issues/50445, the application of `AddDate(0, 0, 0)` is a footgun and does not operate as expected by the code in this package in that it alters the timezone details.

Given the use of period as a representation of duration, we should thus be validating that there is a value to be added for the years, months, and/or days before applying it in order to prevent uninteded consequences where a time period (`PTX`) is being applied.

Amending the arithmetic tests with:

```go
est := mustLoadLocation("America/New_York")
	aest := mustLoadLocation("Australia/Sydney")

	times := []time.Time{
		// A conveniently round number but with non-zero nanoseconds (14 July 2017 @ 2:40am UTC)
		time.Unix(1500000000, 1).UTC(),
		// This specific time fails for EST due behaviour of time.Time.AddDate
		time.Date(2020, 11, 1, 1, 0, 0, 0, est),
		// Testing to ensure that adding zero with a DST day does not blow up due behaviour of time.Time.AddDate
		time.Date(2024, 4, 6, 15, 0, 0, 0, time.UTC).In(aest),
	}

```

When running the tests we get:

```shell
=== NAME  Test_AddTo
    arithmetic_test.go:141:
        0 value=P0D t0=2024-04-07T02:00:00+11:00, t1=2024-04-07T02:00:00+10:00, exp=2024-04-07T02:00:00+11:00
        Expected
            <bool>: false
        to be true
=== NAME  Test_AddTo/0_P0D#02
    testing.go:1576: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
--- FAIL: Test_AddTo (0.00s)
```

Note: the test case included a NYC timestamp that appears to be fine during the transition.

--- a/arithmetic.go
+++ b/arithmetic.go
@@ -6,8 +6,9 @@ package period
 
 import (
 	"errors"
-	"github.com/govalues/decimal"
 	"time"
+
+	"github.com/govalues/decimal"
 )
 
 // AddTo adds the period to a time, returning the result.
@@ -20,13 +21,11 @@ import (
 //
 // However, when years, months or days contains fractions, the result is only an approximation (it
 // assumes that all days are 24 hours and every year is 365.2425 days, as per Gregorian calendar rules).
+//
+// Note: the use of AddDate has unintended consequences when considering the addition of a time only
+// period. See <https://x.com/joelmcourtney/status/1803301619955904979>.
 func (period Period) AddTo(t time.Time) (time.Time, bool) {
-	wholeYears := period.years.Scale() == 0
-	wholeMonths := period.months.Scale() == 0
-	wholeWeeks := period.weeks.Scale() == 0
-	wholeDays := period.days.Scale() == 0
-
-	if wholeYears && wholeMonths && wholeWeeks && wholeDays {
+	if !zeroCalendarValues(period) && wholeCalendarValues(period) {
 		// in this case, time.AddDate provides an exact solution
 
 		years, _, ok1 := period.years.Int64(0)
@@ -233,6 +232,25 @@ func fieldDuration(field decimal.Decimal, factor int64) (int64, bool) {
 	}
 
 	return int64(field.Sign()) * int64(field.Coef()) * factor, factor > 0
+}
+
+func wholeCalendarValues(period Period) bool {
+	zeroCalValue := zeroCalendarValues(period)
+	wholeYears := period.years.Scale() == 0
+	wholeMonths := period.months.Scale() == 0
+	wholeWeeks := period.weeks.Scale() == 0
+	wholeDays := period.days.Scale() == 0
+
+	return !zeroCalValue && wholeYears && wholeMonths && wholeWeeks && wholeDays
+}
+
+func zeroCalendarValues(period Period) bool {
+	zeroYears := period.years == decimal.Zero
+	zeroMonths := period.months == decimal.Zero
+	zeroWeeks := period.weeks == decimal.Zero
+	zeroDays := period.days == decimal.Zero
+
+	return zeroYears && zeroMonths && zeroWeeks && zeroDays
 }
 
 const (

--- a/arithmetic_test.go
+++ b/arithmetic_test.go
@@ -6,11 +6,12 @@ package period
 
 import (
 	"fmt"
-	"github.com/govalues/decimal"
-	. "github.com/onsi/gomega"
 	"math"
 	"testing"
 	"time"
+
+	"github.com/govalues/decimal"
+	. "github.com/onsi/gomega"
 )
 
 func Test_Add_Subtract(t *testing.T) {
@@ -82,12 +83,15 @@ func Test_AddTo(t *testing.T) {
 	const hour = 60 * minute
 
 	est := mustLoadLocation("America/New_York")
+	aest := mustLoadLocation("Australia/Sydney")
 
 	times := []time.Time{
 		// A conveniently round number but with non-zero nanoseconds (14 July 2017 @ 2:40am UTC)
 		time.Unix(1500000000, 1).UTC(),
 		// This specific time fails for EST due behaviour of time.Time.AddDate
 		time.Date(2020, 11, 1, 1, 0, 0, 0, est),
+		// Testing to ensure that adding zero with a DST day does not blow up due behaviour of time.Time.AddDate
+		time.Date(2024, 4, 6, 15, 0, 0, 0, time.UTC).In(aest),
 	}
 
 	for _, t0 := range times {

